src/library_int53.js

```diff
@@ -83,18 +83,32 @@ mergeInto(LibraryManager.library, {
   // converts it to a JavaScript Number, which can represent 53 integer bits precisely.
   // TODO: Add $readI53FromI64Signaling() variant.
   $readI53FromI64: function(ptr) {
```

Comment:
Since I wrote these changes, and it needs review from someone else, can you just remove them from this PR? I'll open a separate PR for them.
They're all just assertion changes so they don't block the rest of this PR.

Reply:
#16879

Reply:
Make sense as the get mapped fix didn't use convertU32PairsToI53 eventually. Removed.
FYI besides rebasing, the clearBuffer method needs updates.

Reply:
I think we still want to replace makeGetU64 with readI53FromU64, makeU64ToNumber with convertU32PairToI53, and either remove or rewrite makeU64ToNumberWithSentinelAsUndefined.

But both that and the clearBuffer change can happen later since this PR is getting close to ready to land.